src/azure-cli/azure/cli/command_modules/monitor/validators.py

```diff
@@ -7,6 +7,16 @@
 from azure.cli.core.commands.validators import validate_tags, get_default_location_from_resource_group
 
 
+def make_cluster_creation_as_no_wait_by_default(cmd, namespace):
+    get_default_location_from_resource_group(cmd, namespace)
+
+    # Make cluster creation no wait by default before ADX cluster can be provisioned automatically.
+    # TODO: remove this method and make cluster creation support both wait and no wait
+    #  after ADX cluster can be provisioned automatically.
+    cmd.supports_no_wait = True
```

Comment:
should we remove this one? If user input `--no-wait False`, will it cause endless loop?

Reply:
No, we'll not expose --no-wait to users as I didn't add supports_no_wait=True in command definition.

---
In reply to: [413669222](https://github.com/Azure/azure-cli/pull/13133#discussion_r413669222) [](ancestors = 413669222)